[Lee Zhi YI] iP

zhiyi12345 opened the pull request:
# Chat with Bobby 

> "Keep calm and carry on!"

Feeling bored, too many things to remember? You can now chat with Bobby!! It's,

- always there
- easy to use
- ~~FAST~~ TOO FAST

All you need to do is, 

1. Download it from [here](https://github.com/zhiyi12345/ip/releases)
2. double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 🤩

AND ITS **FREE**!

Features:

- [x] Manage Tasks
- [x] Manage Deadlines
- [x] Manage Events

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

zhiyi12345 committed:
Level 0
Renamed chatbot to Bobby
Added Greetings and Exit

zhiyi12345 committed:
Level 1, created an echo feature for the chatbot to echo what is inputed by the user unless bye is replied

zhiyi12345 committed:
Level 2, ability to add whatever that is inputed by user except bye and list which returns the exit statement and list of items respectfully
Changed from using scan.next to scan.nextLine to read the whole line

zhiyi12345 committed:
Added in a new class Task, edited Bobby Class to use Task class.
Only to do box added, Mark/unmarked not done yet

zhiyi12345 committed:
Level 3; Mark As Done, Chatbot can now mark and unmark Task

zhiyi12345 committed:
Level 4; created new child classes of Task (events, deadlines, todos)

zhiyi12345 committed:
A TextUiTesting; Fixed some problems resulting from white spaces, and how i want the chatbot new lines to be

zhiyi12345 committed:
Level 5: error handling, handle exceptions for wrong task formats and unknown actions

zhiyi12345 committed:
level 6; created a delete task function, added in a few more error handling for mark and unmark

zhiyi12345 committed:
Added Enum, added error handling for mark and unmark

zhiyi12345 committed:
Add Save function to Bobby the Chatbot

zhiyi12345 committed:
Clear task.txt file

zhiyi12345 committed:
Add a branching.txt file for testing of merge function

zhiyi12345 committed:
Merge branch 'branch-level-7'

zhiyi12345 committed:
Add DateTime to Chatbot
Handle error when input more than bye/list

zhiyi12345 committed:
Delete Branching.txt

zhiyi12345 committed:
Merge branch 'branch-level-8'

zhiyi12345 committed:
Create 2 new classes, TaskList and Storage
Handle exception for actionType

zhiyi12345 committed:
Create a new Parser Class

zhiyi12345 committed:
Merge branch 'add-gradle-support'

zhiyi12345 committed:
Add Java Doc to Bobby class

zhiyi12345 committed:
All JavaDoc to all classes

zhiyi12345 committed:
Correct code to follow Coding Standard

zhiyi12345 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Bobby.java
#	src/main/java/Parser.java

zhiyi12345 committed:
Add find task function to chatbot
Java doc added to new code

zhiyi12345 committed:
Revert "Add find task function to chatbot"

This reverts commit 4757eade13a0879611cd43b6fbc16c254241680b.

zhiyi12345 committed:
Add find function to chatbot, include java doc

zhiyi12345 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Bobby.java

zhiyi12345 committed:
Add JUnit Test to Deadlines and ToDos class

zhiyi12345 committed:
Put all classes into one package bobby.ui

zhiyi12345 committed:
Package App as a Jar file

nuagetsu commented on src/test/java/DeadlinesTest.java:
```diff
@@ -0,0 +1,67 @@
+ * This class ensures that the methods in the Deadlines class
+ * work correctly for task initialization, marking as done,
+ * unmarking, and formatting for display and storage.
+ */
```
Nice writing of clear javadocs to indicate what the class does.

nuagetsu commented on src/test/java/ToDosTest.java:
```diff
@@ -0,0 +1,43 @@
+    ToDos todo1 = new ToDos("Read book");
+
+    @Test
+    public void statusInitializeTest1() {
```
Should there be header comments for each method to give a short description of the test?

nuagetsu commented on src/main/java/Bobby.java:
```diff
@@ -319,28 +320,28 @@ private static void handle_event() throws Exception {
             case deadline:
-                handle_deadline();
+                handleDeadline();
                 break;
```
Should the "case" statements be indented at the same level as the switch statement to follow the coding convetion?

nuagetsu commented on src/main/java/Bobby.java:
```diff
@@ -12,13 +13,13 @@ public class Bobby {
      * Greeting message displayed when the bot starts
      */
-    private static String greeting = "Hello! I'm Bobby\n"
+    private static final String GREETING = "Hello! I'm Bobby\n"
```
Should the name of the GREETING String be changed to START to be similar to the EXIT String in terms of naming due to their similar functions?

nuagetsu commented on src/main/java/Bobby.java:
```diff
@@ -196,7 +197,7 @@ private static void handle_mark() throws Exception {
+    private static void handleUnmark() throws Exception {
         Task curr_task;
         if (scan.hasNextInt()) {
             curr_task = taskList.retrive_task(scan.nextInt());
```
Should the retrieve_task method be written in camel case instead to follow the coding convention?

nuagetsu reviewed (commented):
LGTM, just some nits about coding convention and naming.

itsnotsherm commented on src/main/java/bobby/ui/Bobby.java:
```diff
@@ -0,0 +1,386 @@
+     * @param action The type of action to perform, as determined by user input.
+     */
+    private static void check_action(ActionType action) throws Exception {
+        switch (action) {
```
Should your case clauses not have indentations with regards to the switch statement?

itsnotsherm commented on src/main/java/bobby/ui/Events.java:
```diff
@@ -0,0 +1,48 @@
+    /**
+     * The end time of the Event
+     */
+    protected LocalDateTime end;
```
Should your start and end variables be classified together under one block instead?

itsnotsherm commented on src/main/java/bobby/ui/TaskList.java:
```diff
@@ -0,0 +1,118 @@
+     *
+     * @param input The {@link Task} object to add.
+     */
+    static void add_task(Task input) {
```
Should your method names in this class be written in camelCase?

itsnotsherm commented on src/main/java/bobby/ui/Bobby.java:
```diff
@@ -0,0 +1,386 @@
+     *
+     * @throws Exception if there is an issue writing to the file.
+     */
+    private static void handleBye() throws Exception {
```
I like how you name your methods to handle each command

itsnotsherm commented on src/main/java/bobby/ui/Bobby.java:
```diff
@@ -0,0 +1,386 @@
+            case event:
+                handleEvent();
+                break;
+            case retry:
```
Would it be better to remove this case if there is nothing to execute under its code block?

itsnotsherm commented on src/main/java/bobby/ui/Storage.java:
```diff
@@ -0,0 +1,79 @@
+    /**
+     * File path to store task data
+     */
+    private static String filePath= null;
```
Perhaps you can add a space in between filePath and =?

itsnotsherm reviewed (commented):
LGTM, just a few nits to fix to comply with the java coding standard

zhiyi12345 committed:
Correct code according to CheckStyle

lim-jt commented on src/main/java/bobby/ui/Bobby.java:
```diff
@@ -0,0 +1,366 @@
+     *      or null if the input is invalid
+     */
+
+    public static Task constructTask(String desc) {
```
Well-organised code! Perhaps you could further encapsulate this function into another class i.e. Parser that deals with making sense of the users commands?

lim-jt commented on src/main/java/bobby/ui/Bobby.java:
```diff
@@ -0,0 +1,366 @@
+            }
+            return task;
+        default:
+            System.out.println("GG!! your code is buggy");
```
Haha! While I think this is funny, perhaps it would be more useful to have a more descriptive error message i.e. The task type is invalid?

lim-jt commented on src/main/java/bobby/ui/Bobby.java:
```diff
@@ -0,0 +1,366 @@
+     *
+     * @param action The type of action to perform, as determined by user input.
+     */
+    private static void check_action(ActionType action) throws Exception {
```
Perhaps this could be camelcase to be more in line with naming conventions for methods? Something like checkAction

lim-jt commented on src/main/java/bobby/ui/Bobby.java:
```diff
@@ -0,0 +1,366 @@
+    /**
+     * Scanner to read user input
+     */
+    private static Scanner scan = new Scanner(System.in);
```
Perhaps could rename scan to scanner for clarity and consistency with naming conventions?

lim-jt commented on src/main/java/bobby/ui/Bobby.java:
```diff
@@ -0,0 +1,366 @@
+    /**
+     * Greeting message displayed when the bot starts
+     */
+    private static final String GREETING = "Hello! I'm Bobby\n"
```
Maybe the constants GREETING and EXIT could be renamed to something more descriptive, like GREETING_MESSAGE and EXIT_MESSAGE?

lim-jt reviewed (commented):
LGTM, just need to fix a few nits

zhiyi12345 committed:
Built GUI without Bobby application and without Scene Builder

zhiyi12345 committed:
Implement Bobby with GUI

zhiyi12345 committed:
Add assertion to Bobby and Storage class

Assertion were added to several key methods to ensure program
correctness and catch potential logical errors during execution.

Key changes:
- Added assertions to check if the task format stored in local disk is
in the correct format.

These assertions will throw errors in development environments if any
unexpected conditions are encountered, making debugging easier.

zhiyi12345 committed:
Improve code logic

Some error in handling how tasks are saved to disk. Tasks were not saved
to disk if program terminates improperly.

Rewrites to local storage every time task is added, deleted, marked and
unmarked

zhiyi12345 committed:
Update Javadoc for all the classes

zhiyi12345 committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/bobby/ui/Storage.java

zhiyi12345 committed:
Revert "Update Javadoc for all the classes"

This reverts commit 39f2095d8677f101e7184e194622b34e6bdf8d24.

zhiyi12345 committed:
Revert "Revert "Update Javadoc for all the classes""

This reverts commit 778e9780771774da3a620174692f942b8709acd7.

zhiyi12345 committed:
Add duplicate task detection for deadlines

- Implemented three method to check for duplicate deadlines by comparing
task descriptions and time.
- Improved task management by preventing the addition of duplicate tasks
with identical descriptions and due dates.

zhiyi12345 committed:
Merge pull request #1 from zhiyi12345/branch-C-DetectDuplicates

Add duplicate task detection for deadlines

zhiyi12345 committed:
Improve code logic to ensure event start time is before end time.

zhiyi12345 committed:
Update gradle javaFxVersion

zhiyi12345 committed:
Change profile picture

zhiyi12345 committed:
Remove commented code

zhiyi12345 committed:
Improve code logic to save data to disk

zhiyi12345 committed:
Update README.md

zhiyi12345 committed:
Update README.md

zhiyi12345 committed:
Improve code quality

zhiyi12345 committed:
Merge branch 'master' of https://github.com/zhiyi12345/ip

zhiyi12345 committed:
Update README.md

zhiyi12345 committed:
Update README.md

zhiyi12345 committed:
Add ui.png

zhiyi12345 committed:
Improve code logic for data saving

zhiyi12345 committed:
Update README.md

zhiyi12345 committed:
Update README.md

zhiyi12345 committed:
Merge branch 'master' of https://github.com/zhiyi12345/ip

zhiyi12345 committed:
Update README.md

zhiyi12345 committed:
Update README.md

zhiyi12345 committed:
Update README.md

zhiyi12345 committed:
Update README.md

zhiyi12345 committed:
Improve code logic for find task

zhiyi12345 committed:
Merge branch 'master' of https://github.com/zhiyi12345/ip

zhiyi12345 committed:
Update README.md
